magento/magento2
# Adjust critical CSS extraction and placement

### Description (*)
This pull requests introduces following changes:

- Async CSS loading pattern is updated according to the [newest best practice](https://timkadlec.com/remembers/2020-02-13-when-css-blocks/), allowing us to completely remove `rel="preload"` polyfill.
- Moving existing stylesheets around no longer uses regular expressions which should reduce memory consumption and prevent backtrace limit errors for large pages.
- Styles are now added once, before `</head>` tag, after critical CSS. 

### Related Pull Requests


### Fixed Issues (if relevant)

1. magento/magento2#26498: CSS shows up twice when Critical CSS Enabled

### Manual testing scenarios (*)

1. Enable critical CSS in Magento's admin panel and flush the case.
2. Verify that all external stylesheets are loaded asynchronously without blocking first contentful and meaningful paint.
3. Verify that all external stylesheets are located in the same order at the end of  `<head>`, after inline cricitcal CSS.

### Questions or comments


### Contribution checklist (*)
